observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,217 @@
+        }
+        globalLog = true;
+        globalParams = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+      } else if (methodOrServicePattern.endsWith("/*")) {
+        // TODO(DNVindhya): check if service name is a valid string for a service name
+        // parse config for a service, e.g. "service/*"
+        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
+        if (perServiceFilters.containsKey(service)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
+          continue;
+        }
+        FilterParams params = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+        perServiceFilters.put(service, params);
+      } else {
+        // TODO(DNVVindhya): check if methodOrServicePattern is a valid full qualified method name
+        // parse pattern for a fully qualified method, e.g "service/method"
+        if (perMethodFilters.containsKey(methodOrServicePattern)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
+          continue;
+        }
+        FilterParams params = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+        perMethodFilters.put(methodOrServicePattern, params);
+      }
+    }
+    this.globalLog = globalLog;
+    this.globalParams = globalParams;
+    this.perServiceFilters = ImmutableMap.copyOf(perServiceFilters);
+    this.perMethodFilters = ImmutableMap.copyOf(perMethodFilters);
+    this.methodOrServiceFilterPresent = true;
+  }
+
+  @VisibleForTesting
+  void setEventFilterSet() {
+    List<EventType> eventFilters = config.getEventTypes();
+    if (eventFilters == null) {
+      return;
+    }
+    this.logEventTypeSet = ImmutableSet.copyOf(eventFilters);
+  }
+
+  /**
+   * Class containing results for method/service filter information, such as flag for logging
+   * method/service and payload limits to be used for filtering.
+   */
+  @AutoValue
+  public abstract static class FilterParams {
+    abstract boolean log();
+
+    abstract int headerBytes();
+
+    abstract int messageBytes();
+
+    @VisibleForTesting
+    public static FilterParams create(boolean log, int headerBytes, int messageBytes) {
+      return new AutoValue_ConfigFilterHelper_FilterParams(
+          log, headerBytes, messageBytes);
+    }
+  }
+
+  /**
+   * Checks if the corresponding service/method passed needs to be logged as per the user provided
+   * configuration.
+   *
+   * @param method the fully qualified name of the method
+   * @return MethodFilterParams object 1. specifies if the corresponding method needs to be logged
+   *     (log field will be set to true) 2. values of payload limits retrieved from configuration
+   */
+  public FilterParams isMethodToBeLogged(MethodDescriptor<?, ?> method) {
+    FilterParams params = NO_FILTER_PARAMS;
+    String fullMethodName = method.getFullMethodName();
```

Comment:
Why not declare and get it only when you need it? before line 185. Just like you get `serviceName` on line 188 only when you absolutely need it.